pkg/controllers/management/auth/user_test.go

```diff
@@ -0,0 +1,720 @@
+					prtbMock.EXPECT().Delete("testprtb2", &metav1.DeleteOptions{}).Return(nil),
+				)
+			},
+			expectedError: false,
+		},
+		{
+			name: "error deleting namespaced prtb",
+			inputPRTB: []*v3.ProjectRoleTemplateBinding{
+				&v3.ProjectRoleTemplateBinding{
+					UserName: "testuser",
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "testprtb",
+						Namespace: "testprtbns",
+					},
+				},
+			},
+			mockSetup: func() {
+				prtbMock.EXPECT().DeleteNamespaced("testprtbns", "testprtb", &metav1.DeleteOptions{}).Return(fmt.Errorf("error"))
+			},
+			expectedError: true,
+		},
+		{
+			name: "error deleting prtb",
+			inputPRTB: []*v3.ProjectRoleTemplateBinding{
+				&v3.ProjectRoleTemplateBinding{
+					UserName: "testuser",
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "testprtb",
+					},
+				},
+			},
+			mockSetup: func() {
+				prtbMock.EXPECT().Delete("testprtb", &metav1.DeleteOptions{}).Return(fmt.Errorf("error"))
+			},
+			expectedError: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.mockSetup()
+
+			err := ul.deleteAllPRTB(tt.inputPRTB)
+
+			if tt.expectedError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_deleteUserNamespace(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	namespaceMock := NewMockNamespaceInterface(ctrl)
+	namespaceListerMock := NewMockNamespaceLister(ctrl)
+
+	ul := &userLifecycle{
+		namespaces:      namespaceMock,
+		namespaceLister: namespaceListerMock,
+	}
+
+	tests := []struct {
+		name          string
+		username      string
+		mockSetup     func()
+		expectedError bool
+	}{
+		{
+			name:     "delete namespace",
+			username: "testuser",
+			mockSetup: func() {
+				gomock.InOrder(
+					namespaceListerMock.EXPECT().Get("", "testuser").Return(&v12.Namespace{}, nil),
+					namespaceMock.EXPECT().Delete("testuser", &metav1.DeleteOptions{}).Return(nil),
+				)
+			},
+			expectedError: false,
+		},
+		{
+			name:     "error getting namespace",
+			username: "testuser",
+			mockSetup: func() {
+				gomock.InOrder(
+					namespaceListerMock.EXPECT().Get("", "testuser").Return(nil, fmt.Errorf("")),
+				)
+			},
+			expectedError: true,
+		},
+		{
+			name:     "error deleting namespace",
+			username: "testuser",
+			mockSetup: func() {
+				gomock.InOrder(
+					namespaceListerMock.EXPECT().Get("", "testuser").Return(&v12.Namespace{}, nil),
+					namespaceMock.EXPECT().Delete("testuser", &metav1.DeleteOptions{}).Return(fmt.Errorf("")),
+				)
+			},
+			expectedError: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.mockSetup()
+
+			err := ul.deleteUserNamespace(tt.username)
+
+			if tt.expectedError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_deleteUserSecret(t *testing.T) {
```

Comment:
Could you add a case for when getting the secret returns "NotFound"?

Reply:
sure